drivers/spi/spi_ll_stm32.c

```diff
@@ -743,8 +754,11 @@ static int transceive_dma(const struct device *dev,
 			break;
 		}
 
+#if !defined(CONFIG_SOC_SERIES_STM32H7X)
+		/* toggle the DMA request to restart the transfer */
 		LL_SPI_EnableDMAReq_RX(spi);
 		LL_SPI_EnableDMAReq_TX(spi);
+#endif /* ! CONFIG_SOC_SERIES_STM32H7X */
```

Comment:
Naive question: Can we apply the H7 behavior to all series ?

Reply:
It seems so. 
The Ref Man of the stm32H7 explicitly requires not to change the DMAReq after the SPI enabled. On other series the driver does not follow the recommended sequence (ex. of stm32G4): 
```
When starting communication using DMA, to prevent DMA channel management raising
error events, these steps must be followed in order:
1. Enable DMA Rx buffer in the RXDMAEN bit in the SPI_CR2 register, if DMA Rx is
used.
2. Enable DMA streams for Tx and Rx in DMA registers, if the streams are used.
3. Enable DMA Tx buffer in the TXDMAEN bit in the SPI_CR2 register, if DMA Tx is used.
4. Enable the SPI by setting the SPE _bit.
```
(We see that step 4 comes before step 1)

There the DMA Tx/Rx request bit toggling is used by the driver to control the DMA transfer (assuming the stm32H7 just relies on the spi_dma_move_buffers to progress in the DMA transfer as DMA Tx/Rx request does not change during the DMA transfer).